R/utils.R

```diff
@@ -46,3 +46,38 @@ include_parent_datanames <- function(dataname, join_keys) {
 
   return(unique(c(parents, dataname)))
 }
+
+#' Template function to generate reporter card.
+#'
+#' This function generates a report card with a title,
+#' an optional description, and the option to append the filter state list.
+#'
+#' @param title (`character(1)`) title of the card (unless overwritten by label)
+#' @param label (`character(1)`) label provided by the user when adding the card
+#' @param description (`character(1)`) optional additional description
+#' @param with_filter (`logical(1)`) flag indicating to add filter state
+#' @param filter_panel_api (`FilterPanelAPI`) object with API that allows the generation
+#' of the filter state in the report
+#'
+#' @return (`TealReportCard`) populated with a title, description and filter state
+#'
+#' @export
+card_template <- function(title, label, description = NULL, with_filter, filter_panel_api) {
```

Comment:
Hey @kartikeyakirar what do you think about potential other idea on the name of this function?
- `report_card_template()`
- `structure_report_card()` / `report_card_structure()`
- `compose_report_card()` / `report_card_compose`
??

Also, just a loose idea, since we moved this template to `teal`, and `teal` contains `TealReportCard`. Maybe it would be convenient to create an R6 method in this class that returns the template? `teal::TealReportCard$new()$template(title, label, description = NULL, with_filter, filter_panel_api)`?

This way all `teal.reporter` related functionalities are binded within one class.

Reply:
>  potential other idea on the name of this function?

 Consider the specific functionality  and primary purpose to provide a template for a report card,` report_card_template()` is a good choice. 

> create an R6 method in this class that returns the template?

it is a general-purpose function and doesn't offer a comprehensive structure or template on its own. Users would still need to make additional edits. Therefore, I'm inclined to recommend against incorporating it into TealReportCard. Doing so might lead to confusion among users who wonder why other methods, such as `append_src` and `append_encodings`, .

Reply:
agreed. let's rename to `report_card_template()` and let's do not include in `TealReportCard` for now.